File: 백준/Bronze/25629. 홀짝 수열/홀짝 수열.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.lang.Math;
public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n=Integer.parseInt(br.readLine());
        int rlt[] = new int[2];
        int arr[]=Arrays.stream(br.readLine().split(" "))
        .mapToInt(Integer::parseInt).toArray();
        for(int i=0;i<n;i++) {
            if(arr[i]%2!=0)
                rlt[0]+=1;
            else
                rlt[1]+=1;
        }
        System.out.print(rlt[0]==Math.ceil(n/2.0) && rlt[1]==n/2?1:0);
    }
}

File: 백준/Bronze/25629. 홀짝 수열/홀짝 수열.py
import sys,math
input=lambda:sys.stdin.readline().rstrip()
n,rlt=int(input()),[0]*2
arr=[*map(int,input().split())]
for i in range(n):
    if arr[i]%2!=0:
        rlt[0]+=1
    else:
        rlt[1]+=1
print(1 if rlt[0]==math.ceil(n/2) and rlt[1]==n//2 else 0)